fn/req.go

```diff
@@ -0,0 +1,31 @@
+package fn
+
+// Req is a type to encapsulate RPC-like calls wherein we send some data
+// structure as a request, as well as a channel to receive the response on where
+// the remote goroutine will send the result.
+type Req[Input any, Output any] struct {
+	// Request is the data we are sending to the remote goroutine for
+	// processing.
+	Request Input
+
+	// Response is the channel on which we will receive the result of the
+	// remote computation.
+	Response chan<- Output
+}
+
+// NewReq is the base constructor of the Req type. It returns both the packaged
+// Req object as well as the receive side of the response channel that we will
+// listen on for the response.
+func NewReq[Input, Output any](input Input) (
+	Req[Input, Output], <-chan Output) {
+
+	// Always buffer the response channel so that the goroutine doing the
+	// processing job does not block if the original requesting routine
+	// takes an unreasonably long time to read the response.
+	responseChan := make(chan Output, 1)
```

Comment:
I wouldn't mind just having a constructor with `input, buf_size` fwiw, not too much boilerplate and makes the choice more conscious.

Reply:
Can either of you give an example of a use case where a request will have *multiple* responses of the `Output` type? Intuitively this seems odd to me as it breaks away from a request/response pattern and then we would also need to know when the response stream is terminated.

If we have a need for multi-response I'd rather just create another abstraction for something like that unless there is a sensible generalization of the new use cases. Lifting the parameter out is fine, but it leaks internals in a way I don't think is good.

Reply:
We do have a couple of RPCs in `lnd` that take a single request and then return multiple responses. Not exactly the same, I know. I just want to avoid this being used for something it shouldn't. So maybe just a sentence in the Godoc is enough, something along the lines of "The channel is buffered with a size of 1, so this is only intended to be used for single responses".

Reply:
> I actually think the choice being "more conscious" is precisely what I want to avoid

Fair enough! I though we had a few places with buffer > 1, but took a more thorough look and that's mostly in tests / not for response-requests.

Reply:
I added a warning label to the godoc and also added some nice convenience methods on top.